airflow/providers/amazon/provider.yaml

```diff
@@ -288,6 +288,7 @@ operators:
   - integration-name: AWS Lambda
     python-modules:
       - airflow.providers.amazon.aws.operators.aws_lambda
```

Comment:
Is old entry needed? I think, it is only used for documentation.

Reply:
Removing it cause test failures.
We can remove the entry once we remove the operator

Reply:
Hmm. I think we should fix the failing tests actually. I prefer to keep the provider.yaml to describe "current" state not some mixture between and old state - we do not track which entries are deprecated here. 

BTW. Maybe there are other cases like that already but if they are, we should fix it so that the tests do not fail here and right after we should add tests finding similar deprecated entries and remove them. Happy to do it as a follow-up actually.

Reply:
I think this better to be handled in a different PR. The provider.yaml is full of such deprecated records

Reply:
Really. Haven't checked? That's bad then. Let's do it in separate PR then.